docs/security-considerations.rst

```diff
@@ -657,3 +657,14 @@ Notice that we do not clear knowledge about ``array`` and ``d`` because they
 are located in storage, even though they also have type ``uint[]``.  However,
 if ``d`` was assigned, we would need to clear knowledge about ``array`` and
 vice-versa.
+
+Real World Assumptions
+======================
+
+Some scenarios are expressive in Solidity and the EVM, but are expected to
+never occur in practice.
+One of such cases is the length of a dynamic storage array overflowing after a
+push: If the ``push`` operation is applied to such array 2^256 times, its
```

Comment:
This assumes that `push` is the only way to change the length, but it can also be done by assignment.